### PR TITLE
Machine network performance measurement

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Ping time, machine internet quality measurement](https://trello.com/c/xKiQCLCO/768-ping-time-machine-internet-quality-measurement)

File: lib/brain.js [hunk 1]
```diff
@@ -131,6 +132,18 @@ const Brain = function (config) {
   this.powerDown = false
   this.beforeIdleState = true
   this.scannerTimeout = null
+  this.pingInterval = null
+  this.speedtestInterval = null
+
+  pingInterval = setInterval(() => {
+    ping.pingRepository()
+      .then(arr => this.trader.networkHeartbeat(arr))
```
Comment (jsousa127) [inline lib/brain.js:140]: `.then(this.trader.networkHeartbeat)`

File: lib/brain.js [hunk 2]
```diff
@@ -131,6 +132,18 @@ const Brain = function (config) {
...
+  }, this.config.pingInterval)
+
+  speedtestInterval = setInterval(() => {
+    ping.checkDownloadSpeed(10)
+      .then(arr => this.trader.networkPerformance(arr))
```
Comment (jsousa127) [inline lib/brain.js:145]: `.then(this.trader.networkPerformance)`

File: lib/trader.js [hunk 3]
```diff
@@ -480,6 +480,24 @@ Trader.prototype.pollError = function pollError (err) {
   this.emit('networkDown')
 }
 
+Trader.prototype.networkHeartbeat = function networkHeartbeat (obj) {
+  return this.request({
+    path: '/network/heartbeat',
+    body: obj,
+    method: 'POST'
+  })
+    .catch(err => console.log('Failed to send network heartbeat', err))
```
Comment (jsousa127) [inline lib/trader.js:489]: Maybe it's better to use `console.error`, what do you think?

**Review by jsousa127 — CHANGES_REQUESTED** [review]
Looking good in general, added a few comments about some small details

File: lib/trader.js (hunk 3 above)
Comment (chaotixkilla) [reply lib/trader.js:489]: Correct

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**